Patch-Source: https://gitlab.alpinelinux.org/postmarketOS/alpine-appstream-downloader/-/commit/c8e0e83b062da880834efba7d53ede3a42d0dd10
---
From c8e0e83b062da880834efba7d53ede3a42d0dd10 Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Sat, 28 Sep 2024 15:18:58 +0200
Subject: [PATCH] Use tarfile.extractall(filter=) argument

Since Python 3.12+, tarfile.extractall() supports extraction filters.

We should explicitly set the filter to resolve the DeprecationWarning:

/usr/sbin/alpine-appstream-downloader:93: DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior.

This drops support for Python 3.11 and below.

Ref: https://docs.python.org/3.14/library/tarfile.html#extraction-filters
---
 alpine-appstream-downloader | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/alpine-appstream-downloader b/alpine-appstream-downloader
index 4e96168..b4be05b 100755
--- a/alpine-appstream-downloader
+++ b/alpine-appstream-downloader
@@ -91,7 +91,8 @@ def setup_appstream_data(release: str, repo: str):
         r = get(icon_url, stream=True)
         icons = tarfile.open(fileobj=r.raw, mode='r')
         origin = ORIGIN_FORMAT.format(release, repo)
-        icons.extractall(f'{AS_BASE_FOLDER}/icons/{origin}/{size}')
+        icons.extractall(f'{AS_BASE_FOLDER}/icons/{origin}/{size}',
+                         filter='data')
 
 
 def clean_unused_appstream(repos: list[(str, str)]):
-- 
GitLab

